herbert_core/utilities/ortho_vec.m

```diff
@@ -0,0 +1,18 @@
+function [v1,v2] = ortho_vec(v0)
+% The routine returns two vectors, orthogonal to the given one
+%
+[mv,ind] = max(abs(v0));
+if mv<eps
+    error('HERBERT:ortho_vec:invalid_argument',...
+        'Input vector %s has zero size, so is orthogonal to any vector',...
+        evalc('disp(v0)'));
+end
+
+v2 = ones(size(v0));
+v2(ind) = 0;
+v1 = cross(v0,v2);
+v1 = v1/norm(v1);
+v2 = cross(v0,v1);
+v2 = v2/norm(v2);
```

Comment:
Quite ingeneous. My equivalent solution for v2 is
        `[mvx,indx]=max(abs(v0))`
        `[mvn,indn]=min(abs(v0))`
        `v2 = ones(size(v0))`
        `v2(indx)=0`
        `v2(indn)=0`
        `indm=find(v2,1)`
        `v2 = v0`
        `v2(indx)=v0(indm)`
        `v2(indm)=-mvx`
which is a rotation of v0 about its smallest component direction. Yours is certainly more elegant.

Reply:
interesting. Has advantage of selecting axis aligned orthogonal for axis aligned v0, but needs a lot of thinking to understand if works as intended. i.e. would not work for [1,1,1], though the idea is promising. I will leave my code as it is but feel free to improve appropriately